pilot/docker/envoy_pilot.yaml.tmpl

```diff
@@ -153,17 +153,17 @@ static_resources:
             max_concurrent_streams: 1073741824
           http_filters:
           - config:
-              default_destination_service: istio-pilot.{{ .PodNamespace }}.svc.cluster.local
```

Comment:
if this is only for the sidecars in pilot, mixer, telemetry, etc. then the logic you have is not needed. Simply use name.namespace format

Reply:
what about the case when a specific domain name might be needed? Adding this functionality should not cause any negative either performance or functionality impact, but could be useful.

Reply:
The name of the config is not important, but the content must include the cluster domain in the destination service host. Otherwise, things will break downstream.